src/python/bot/minimizer/antlr_tokenizer.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Antlr Tokenizer"""
+
+from builtins import object
+import antlr4
+
+
+class AntlrTokenizer(object):
+  """Tokenizer. Takes an Antlr Lexer created using
+  $ antlr4 -Dlanguage=Pythonn <AntlrGrammar.g4>
+  and allows user to tokenize files using that grammar"""
+
+  def __init__(self, lexer):
+    """constructor"""
+    self._lexer = lexer
+
+  def fill(self, stream):
+    """Helper function. antlr4.CommonTokenStream.fill should work, but
+    it does not fetch all of the tokens. This is a replacement that works"""
+    i = 0
+    while stream.fetch(1):
+      i += 1
+    return i
```

Comment:
Shouldn't this return the tokens itself and not just the count, then you shouldn't need to do stream.getTokens

Reply:
fetch(x) fills a buffer with x tokens, and stream.getTokens retrieves them from the buffer. Not sure why it works this way either.

Reply:
Ok fine then.